docs/en/advanced_guides/datasets.md

```diff
@@ -1 +1,143 @@
+|                  `self._metainfo`                   |              Set meta information of dataset.               |
+|                `self._join_prefix()`                |            Join paths of images and annotations.            |
+|                   `self.pipeline`                   | Build data pipeline for data preprocessing and preparation. |
+|                 `self.full_init()`                  |                Full initialize the dataset.                 |
+
+In `BaseSegDataset`, the argument `lazy_init` (Defaults to False) is used to control whether to load annotation during instantiation.
+If not using lazy initialization, the `self.full_init()` would be called to execute several methods from top to bottom:
+
+| Methods executed in `self.full_init()` |                           Usage                            |
+| :------------------------------------: | :--------------------------------------------------------: |
+|        `self.load_data_list()`         |                   Load data information.                   |
+|          `self.filter_data()`          | Filter illegal data, such as data that has no annotations. |
+|   `self._get_unserialized_subset()`    |           Get subset data according to indices.            |
+|        `self._serialize_data()`        |   Serialize `data_list` of corresponding dataset class.    |
+
+In some cases such as visualization, only the meta information of the dataset is necessary, while loading annotation file
+is unnecessary. `BaseSegDataset` can skip loading annotations to save time by set `lazy_init=True`, in this case `self.full_init()` would not be executed.
+
+### Loading meta info
+
+Meta information is collected in `self._metainfo` method.
+If `metainfo` contains existed filename path, it will be parsed by `list_from_file`, otherwise it is simply parsed as meta information:
+
+```python
+def _load_metainfo(cls, metainfo: dict = None) -> dict:
+    # avoid `cls.METAINFO` being overwritten by `metainfo`
+    cls_metainfo = copy.deepcopy(cls.METAINFO)
+    ...
+    for k, v in metainfo.items():
+        if isinstance(v, str):
+            # If type of value is string, and can be loaded from
+            # corresponding backend. it means the file name of meta file.
+            try:
+                cls_metainfo[k] = list_from_file(v)
+            except (TypeError, FileNotFoundError):
+                warnings.warn(f'{v} is not a meta file, simply parsed as '
+                              'meta information')
+                cls_metainfo[k] = v
+        else:
+            cls_metainfo[k] = v
+    return cls_metainfo
+```
+
+Then `_load_metainfo` returns parsed meta information for train/test process.
+
+### Loading data info
+
+By default, `self.load_data_list()` would be called in `self.full_init()`. In `BaseSegDataset`,
+`self.load_data_list()` function is overwritten where annotation path `seg_map_path` would be added from annotation file directory or its meta file.
```

Comment:
I think it would be better to add a table for 2 kinds of loading, and says '`load_data_list ` can be overwritten when it doesn't meet your datasets'.